### PR TITLE
feat: added phone number identifier

### PR DESCRIPTION
**Pull request by alexey-reshetnik**
Added phone number identifier support for verification.

## Related issue(s)
#137

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Review by aeneasr — COMMENTED** [review]
Looking pretty good so far! How do you want to handle sending SMS?

**Comment by reshetnik-alexey** [comment]
Created a separate PR for sending sms for easier review: https://github.com/ory/kratos/pull/1941

File: identity/extension_verify.go [hunk 1]
```diff
@@ -26,42 +25,54 @@ func (r *SchemaExtensionVerification) Run(ctx jsonschema.ValidationContext, s sc
...
-		}
+		r.appendAddress(address)
 
-		if has := r.has(r.v, address); has == nil {
-			r.v = append(r.v, *address)
-		}
+		return nil
+
+	case AddressTypePhone:
+		address := NewVerifiablePhoneAddress(fmt.Sprintf("%s", value), r.i.ID)
```
Comment (aeneasr) [inline identity/extension_verify.go:40]: Is there some canonical verification type (e.g. RFC) for this?

**Review by aeneasr — COMMENTED** [review]
This looks good on a high level (will need to dig into the tests still)! Could you also please add some docs for this feature, probably around here https://www.ory.sh/kratos/docs/concepts/email-sms#sending-sms?

**Comment by nicolasburtey** [comment]
this PR/features is awesome. we (https://github.com/GaloyMoney/galoy) will likely switch to Ory once this is done.

**Comment by saltbo** [comment]
Look forward to this update. Come on.

File: identity/extension_verify.go (hunk 1 above)
Comment (alexey-reshetnik) [reply identity/extension_verify.go:40]: I think that we can use some implementation of libphonenumber for validation purposes, for example [one](https://github.com/nyaruka/phonenumbers) that was mentioned in issue. 
But we need additional work to implement validation in proper way, so decided to not include it in this PR.
I'm thinking that we need to extend JSON schema [library](https://github.com/ory/jsonschema) with "phone" type and use the phonenumber validation in there.
We can create a separate issue and implement the validation this way or any other in separate PR, what do you think?

Comment (aeneasr) [reply identity/extension_verify.go:40]: Yes, that sounds good to me! The problem is usually that we can get hung up on this. We have problem with SMTP already where using an invalid email address (e.g. `asdf`) can lead to the SMTP server rejecting the request, which in turn causes these addresses to be retried (currently forever). So I think we could alleviate this problem a bit with validation.

**Comment by mazhenming892** [comment]
come on

**Comment by aeneasr** [comment]
Please follow our code of conduct and community guidelines and refrain from putting pressure on open source contributors. No one is preventing you from doing the work yourself and you have no right to try and force others to do it for you. Thank you.

File: identity/extension_verify.go (hunk 1 above)
Comment (alexey-reshetnik) [reply identity/extension_verify.go:40]: Got it. Okay, will submit PR with phone number support to ory/jsonschema then.

Comment (alexey-reshetnik) [reply identity/extension_verify.go:40]: Created a PR, please review: https://github.com/ory/jsonschema/pull/7 Once it's merged, we can merge this one too, I think

**Comment by alexey-reshetnik** [comment]
Removed PR from draft status, added validation and test. @aeneasr please, review once again

**Comment by aeneasr** [comment]
> Have one problem with failing docker image scanner, but it is related to protobuf package which was not changed in this PR (also you can check out anchore/grype#558, seems like this failing check is false-positive).

Yes, that is indeed a false positive. No problem. The e2e test is also flaky 😅 

How would you like to continue with the other SMS PR that is going on? Do we merge this one, then rebase the other? I have time to review tomorrow :)

**Comment by aeneasr** [comment]
Scanning over the code quickly, this looks good to merge 🎉 

I will do another pass tomorrow but I think this is a no-brainer in terms of merging, great job :)

**Comment by alexey-reshetnik** [comment]
Yep, I am finishing the pull request with the courier now - encountered a few errors

**Comment by vinckr** [comment]
Hello @alexey-reshetnik 
Congrats on merging your first PR in Ory 🎉 !
Your contribution is helping secure millions of identities around the globe 🌏 
As a small token of appreciation we send all our first time contributors a gift package to welcome them to the community. 
[Please drop me an email to claim your Ory swag!](mailto:vincent@ory.sh)

**Pull request merged**